src/screens/tabs/Tokens/components/AddToken/steps/CustomToken/index.tsx

```diff
@@ -0,0 +1,201 @@
+import { t } from 'i18next';
+import React, { useCallback, useMemo, useRef, useState } from 'react';
+import { Keyboard, Linking, View } from 'react-native';
+import { Modalize } from 'react-native-modalize';
+import { useDispatch } from 'react-redux';
+
+import Button from '@/components/buttons/Button';
+import RainbowButton from '@/components/buttons/RainbowButton';
+import {
+  ActionButton,
+  ActionSheet,
+  Header,
+  Modal,
+  Text,
+  TextInput,
+} from '@/components/common';
+import ErrorIcon from '@/components/icons/svg/Error.svg';
+import Info from '@/components/icons/svg/Info.svg';
+import { customTokensUrl } from '@/constants/urls';
+import { DABToken } from '@/interfaces/dab';
+import { FungibleStandard } from '@/interfaces/keyring';
+import { getTokenInfo } from '@/redux/slices/user';
+import { validateCanisterId } from '@/utils/ids';
+
+import styles, { iconColor } from './styles';
+import { Error, getError } from './utils';
+
+interface Props {
+  modalRef: React.RefObject<Modalize>;
+  onSelectedToken: (token: DABToken) => void;
+}
+
+function CustomToken({ modalRef, onSelectedToken }: Props) {
+  const [canisterId, setCanisterId] = useState('');
+  const dispatch = useDispatch();
+  const [standard, setStandard] = useState<FungibleStandard>('DIP20');
```

Comment:
DIP20 should be a value from a enum or something like that

Reply:
same, they are already defined as types in `FungibleStandard`

Reply:
![Screen Shot 2022-08-22 at 12 39 30](https://user-images.githubusercontent.com/25931366/185961764-0e28d688-d92f-4813-994e-a560f9c97947.png)